client/src/main/java/io/pravega/client/stream/impl/ClientFactoryImpl.java

```diff
@@ -213,9 +214,8 @@ public ClientFactoryImpl(String scope, Controller controller, ConnectionPool con
     private <T> RevisionedStreamClient<T> createRevisionedStreamClient(Segment segment, Serializer<T> serializer,
                                                                        SynchronizerConfig config) {
         EventSegmentReader in = inFactory.createEventReaderForSegment(segment, config.getReadBufferSize());
-        String delegationToken = Futures.getAndHandleExceptions(controller.getOrRefreshDelegationTokenFor(segment.getScope(),
-                                                                                                          segment.getStreamName()), RuntimeException::new);
-        DelegationTokenProvider delegationTokenProvider = DelegationTokenProviderFactory.create(delegationToken, controller, segment);
+        DelegationTokenProvider delegationTokenProvider = DelegationTokenProviderFactory.create(controller, segment,
+                AccessOperation.READ);
         ConditionalOutputStream cond = condFactory.createConditionalOutputStream(segment, delegationTokenProvider, config.getEventWriterConfig());
         SegmentMetadataClient meta = metaFactory.createSegmentMetadataClient(segment, delegationTokenProvider);
         return new RevisionedStreamClientImpl<>(segment, in, outFactory, cond, meta, serializer, config.getEventWriterConfig(), delegationTokenProvider);
```

Comment:
This would cause the SegmentOutputStream created by the RevisionedStreamClient to have READ permissions right ?

Reply:
Yes, that is right. That's no issue though as the segment store shall allow writes to internal streams with READ permissions (for the Controller, a [configuration ](https://github.com/pravega/pravega/blob/2798cba7eee05aa112af92fce43b21b470c5c683/controller/src/main/java/io/pravega/controller/util/Config.java#L116)determines whether it is allowed or not):

https://github.com/pravega/pravega/blob/2798cba7eee05aa112af92fce43b21b470c5c683/segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java#L150